core/apple/test/SourceNSInputStreamTest.kt

```diff
@@ -0,0 +1,354 @@
+        }
+    }
+
+    @Test
+    fun bufferInputStreamLongData() {
+        val source = Buffer()
+        source.writeString("a" + "b".repeat(Segment.SIZE * 2) + "c")
+        testInputStreamLongData(source.asNSInputStream())
+    }
+
+    @Test
+    fun realSourceInputStreamLongData() {
+        val source = Buffer()
+        source.writeString("a" + "b".repeat(Segment.SIZE * 2) + "c")
+        testInputStreamLongData(RealSource(source).asNSInputStream())
+    }
+
+    private fun testInputStreamLongData(input: NSInputStream) {
+        val lengthPlusOne = Segment.SIZE * 2 + 3
+        val byteArray = ByteArray(lengthPlusOne)
+        byteArray.usePinned {
+            val cPtr = it.addressOf(0).reinterpret<uint8_tVar>()
+
+            assertEquals(NSStreamStatusNotOpen, input.streamStatus)
+            assertEquals(-1, input.read(cPtr, lengthPlusOne.convert()))
+            input.open()
+            assertEquals(NSStreamStatusOpen, input.streamStatus)
+
+            byteArray.fill(-5)
+            assertEquals(Segment.SIZE.convert(), input.read(cPtr, lengthPlusOne.convert()))
+            assertEquals("[97${", 98".repeat(Segment.SIZE - 1)}${", -5".repeat(Segment.SIZE + 3)}]", byteArray.contentToString())
+
+            byteArray.fill(-6)
+            assertEquals(Segment.SIZE.convert(), input.read(cPtr, lengthPlusOne.convert()))
+            assertEquals("[98${", 98".repeat(Segment.SIZE - 1)}${", -6".repeat(Segment.SIZE + 3)}]", byteArray.contentToString())
+
+            byteArray.fill(-7)
+            assertEquals(2, input.read(cPtr, lengthPlusOne.convert()))
+            assertEquals("[98, 99${", -7".repeat(Segment.SIZE * 2 + 1)}]", byteArray.contentToString())
+
+            byteArray.fill(-8)
+            assertEquals(0, input.read(cPtr, lengthPlusOne.convert()))
+            assertEquals("[-8${", -8".repeat(lengthPlusOne - 1)}]", byteArray.contentToString())
+        }
+    }
+
+    @Test
+    fun nsInputStreamClose() {
+        val buffer = Buffer()
+        buffer.writeString("abc")
+        val source = RealSource(buffer)
+        assertFalse(source.closed)
+
+        val input = source.asNSInputStream()
+        input.open()
+        input.close()
+        assertTrue(source.closed)
+        assertEquals(NSStreamStatusClosed, input.streamStatus)
+
+        val byteArray = ByteArray(4)
+        byteArray.usePinned {
+            val cPtr = it.addressOf(0).reinterpret<uint8_tVar>()
+
+            byteArray.fill(-5)
+            assertEquals(-1, input.read(cPtr, 4U))
+            assertNotNull(input.streamError)
+            assertEquals("Underlying source is closed.", input.streamError?.localizedDescription)
+            assertEquals("[-5, -5, -5, -5]", byteArray.contentToString())
+        }
+    }
+
+    @Test
+    fun delegateTest() {
+        val runLoop = startRunLoop()
+
+        fun consumeWithDelegate(input: NSInputStream, data: String) {
+            println("consumeWithDelegate(${input::class}, $data)")
```

Comment:
This was for debugging purposes with the test that was failing in the CI. I'll need to find a better way to track the specific function call that failed if it's not logged, since coroutine stack traces are truncated, you can't see the original caller to know which stream caused the failure.

Reply:
After removing the platform stream tests, most of the functions are only called once. So hopefully there isn't a need for these further debug details.